src/autopas/containers/verletClusterLists/Cluster.h

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file Cluster.h
+ * @author humig
+ * @date 27.07.19
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "autopas/utils/SoAView.h"
+
+namespace autopas::internal {
+
+/**
+ * This class represents a cluster in the VerletClusterLists container.
+ *
+ * It contains a pointer to the particles for AoS, a SoAView for SoA, and the neighbor list for this cluster.
+ *
+ * @tparam Particle The type of the particles this cluster consists of.
+ * @tparam clusterSize The number of particles in the cluster.
+ */
+template <class Particle, size_t clusterSize>
+class Cluster {
+ public:
+  /**
+   * Constructs a cluster starting from firstParticle and going on for clusterSize particles.
+   *
+   * Caller is responsible that there are enough particles after this particle in memory.
+   *
+   * @param firstParticle A pointer to the first particle of the cluster.
+   */
+  explicit Cluster(Particle *firstParticle) : _firstParticle(firstParticle) {}
+
+  /**
+   * Returns the particle at position index in the cluster.
+   *
+   * No index checking is performed!
+   *
+   * @param index The index of the particle to return.
+   * @return the particle at position index in the cluster.
+   */
+  Particle &operator[](size_t index) { return *(_firstParticle + index); }
+
+  /**
+   * @copydoc operator[](size_t)
+   */
+  const Particle &operator[](size_t index) const { return *(_firstParticle + index); }
+
+  /**
+   * Returns the SoAView for this cluster.
+   * @return the SoAView for this cluster.
+   */
+  auto &getSoAView() { return _soaView; }
```

Comment:
getSoAView should actually return an object and not a reference (Views should always be lightweight and be copied)